library/cc/engine_builder.cc

```diff
@@ -89,8 +89,23 @@ EngineSharedPtr EngineBuilder::build() {
     }
   }
 
-  Engine* engine = new Engine(init_engine(), config_str, this->log_level_, this->callbacks_);
-  return EngineSharedPtr(engine);
+  auto envoy_engine = init_engine();
+  Engine* engine = new Engine(envoy_engine);
+  auto engine_ptr = EngineSharedPtr(engine);
+
+  auto callbacks = this->callbacks_.release();
+  callbacks->parent = engine_ptr;
+
+  envoy_logger null_logger{
+      .log = nullptr,
+      .release = envoy_noop_release,
+      .context = nullptr,
+  };
```

Comment:
Would need to verify, but by convention, it should be safe to simply calloc this (or any other envoy mobile struct) if you're not going to use it.